oommfc/oommf/oommf.py

```diff
@@ -299,150 +343,173 @@ def errors(self):
         raise EnvironmentError(msg)
 
 
-def get_oommf_runner(use_cache=True, envvar='OOMMFTCL',
-                     oommf_exe='oommf', docker_exe='docker'):
-    """Find the best available way to run OOMMF.
+class Runner:
+    """Control the default runner."""
 
-    Returns an ``oommfc.oommf.OOMMFRunner`` object, or raises
-    ``EnvironmentError`` if no suitable method is found.
+    def __init__(self):
+        self.cache_runner = True
+        self.envvar = 'OOMMFTCL'
+        self.oommf_exe = 'oommf'
+        self.docker_exe = 'docker'
+        self._runner = None
 
-    Parameters
-    ----------
-    use_cache : bool
+    @property
+    def runner(self):
+        """Return default OOMMF runner."""
+        if self.cache_runner and self._runner is not None:
+            log.debug('Returning ceched runner.')
+            return self._runner
+        self.autoselect_runner()
+        return self._runner
 
-      The first call to this function will determine the best way to run OOMMF
-      and cache it. Normally, subsequent calls will return the ``OOMMFRunner``
-      object from the cache. Setting this parameter to ``False`` will cause it
-      to check for available methods again. Defaults to ``True``.
+    @runner.setter
+    def runner(self, runner):
+        if runner.status == 1:
+            raise ValueError(f'{runner=} cannot be used.')
+        self._runner = runner
 
-    envvar : str
+    def autoselect_runner(self):
```

Comment:
Does the method need to take parameters (which are described below), such as `use_cache`, `oommf_exe` etc?

Alternatively, we can explain in the docstring (see comment below).

Reply:
Looks like you reviewed a version with outdated docstrings. I changed this in the mean time. Please again have a look at the latest version.